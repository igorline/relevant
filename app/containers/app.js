'use strict';

import React, {
  AppRegistry,
  Component,
  StyleSheet,
  Text,
  View,
  TextInput
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as authActions from '../actions/authActions';
var {Router, routerReducer, Route, Container, Animations, Schema} = require('react-native-redux-router');
var {NavBar, NavBarModal} = require('../components/NavBar');
import Auth from './auth';
import Import from './import';
import Profile from './profile';

class Application extends Component {
  render() {
    return (

    <View style={{flex:1}} >
                <View style={{position:'absolute',left:0,right:0,top:0,bottom:0,backgroundColor:'white'}}/>
                <Router>
                    <Schema name="modal" sceneConfig={Animations.FlatFloatFromBottom} navBar={NavBarModal}/>
                    <Schema name="default" sceneConfig={Animations.FlatFloatFromRight} navBar={NavBar}/>
                    <Schema name="withoutAnimation" navBar={NavBar}/>
                    <Schema name="tab" navBar={NavBar}/>

                  <Route name="Auth" component={Auth}  hideNavBar={true} type="replace" initial={true} title="Home" />
                  <Route name="SignUp" component={Auth} hideNavBar={true} type="replace" title="Sign up" />
                  <Route name="LogIn" component={Auth} hideNavBar={true}  type="replace" title="Log in" />
                  <Route name="Import" component={Import} hideNavBar={true}  type="replace" title="Import" />
                  <Route name="Profile" component={Profile} hideNavBar={true}  type="replace" title="Profile" />
                </Router>

            </View>

    );
  }
}

export default Application

