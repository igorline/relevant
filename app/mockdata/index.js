import * as activity from 'app/mockdata/activity';
import * as auth from 'app/mockdata/auth';
import * as community from 'app/mockdata/community';
import * as earnings from 'app/mockdata/earnings';
import * as post from 'app/mockdata/post';
import * as user from 'app/mockdata/user';
import * as twitter from 'app/mockdata/twitter';

export { activity, auth, community, earnings, post, user, twitter };
