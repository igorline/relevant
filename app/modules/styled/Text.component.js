import styled from 'styled-components/primitives';
import { colors, fonts, sizing } from 'app/styles/globalStyles';

export const Header = styled.Text`
  font-size: ${sizing.byUnit(3)};
  line-height: ${sizing.byUnit(3)};
  ${fonts.HelveticaNeueBold}
  color: ${(props) => props.color ? props.color : colors.black};
`;

export const Title = styled.Text`
  font-size: ${sizing.byUnit(2.5)};
  line-height: ${sizing.byUnit(2.5)};
  ${fonts.HelveticaNeueCondensedBold}
  color: ${(props) => props.color ? props.color : colors.black};
`;

export const NumericalValue = styled.Text`
  font-size: ${sizing.byUnit(1.75)};
  line-height: ${sizing.byUnit(1.75)};
  ${fonts.HelveticaNeueCondensedBold}
  color: ${(props) => props.color ? props.color : colors.secondaryText};
`;

export const LinkWithIcon = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.HelveticaNeueMedium}
  color: ${(props) => props.color ? props.color : colors.black};
`;

export const AltLinkTitle = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.HelveticaNeueMedium}
  color: ${(props) => props.color ? props.color : colors.black};
  text-decoration: underline;
`;

export const BodyText = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.Helvetica}
  color: ${(props) => props.color ? props.color : colors.black};
`;

export const CommentText = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.Georgia}
  color: ${(props) => props.color ? props.color : colors.black};
`;

export const SecondaryText = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.Helvetica}
  color: ${(props) => props.color ? props.color : colors.secondaryText};
`;

export const CTALink = styled.Text`
  font-size: ${sizing.byUnit(1.5)};
  line-height: ${sizing.byUnit(1.5)};
  ${fonts.Helvetica}
  color: ${(props) => props.color ? props.color : colors.blue};
`;
