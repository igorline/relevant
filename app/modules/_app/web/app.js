import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import Header from 'modules/navigation/web/header.component';
import AuthContainer from 'modules/auth/web/auth.container';
import * as navigationActions from 'modules/navigation/navigation.actions';
import * as authActions from 'modules/auth/auth.actions';
import { getEarnings } from 'modules/wallet/earnings.actions';
import { getCommunities } from 'modules/community/community.actions';
import AddEthAddress from 'modules/wallet/web/AddEthAddress';
import Modal from 'modules/ui/web/modal';
import CreatePost from 'modules/createPost/createPost.container';
import EthTools from 'modules/web_ethTools/tools.container';
import Eth from 'modules/web_ethTools/eth.context';
import { ToastContainer } from 'react-toastify';
import { GlobalStyle } from 'app/styles';
import * as modals from 'modules/ui/modals';

if (process.env.BROWSER === true) {
  require('app/styles/index.css');
  require('app/styles/fonts.css');
  require('modules/web_splash/splash.css');
  require('react-toastify/dist/ReactToastify.css');
}

class App extends Component {
  static propTypes = {
    auth: PropTypes.object,
    actions: PropTypes.object,
    match: PropTypes.object,
    location: PropTypes.object,
    user: PropTypes.object,
    children: PropTypes.node,
    history: PropTypes.object,
    route: PropTypes.object,
    activeCommunity: PropTypes.string,
    globalModal: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
  };

  state = {
    openLoginModal: false
  };

  componentWillMount() {
    const { community } = this.props.auth;
    if (community && community !== 'home') {
      this.props.actions.setCommunity(community);
    }
  }

  componentDidMount() {
    const { actions, auth } = this.props;
    const { community } = auth;

    actions.setCommunity(community);
    actions.getCommunities();
    actions.getUser();
    actions.getEarnings('pending');

    if (auth.user && !auth.user.webOnboard.onboarding) {
      actions.showModal('onboarding');
      actions.webOnboard('onboarding');
    }
    // TODO do this after a timeout
    // window.addEventListener('focus', () => {
    //   if (this.props.newPosts)
    //   this.props.actions.refreshTab('discover');
    // });
  }

  componentDidUpdate(prevProps) {
    const { actions, auth, location, match, activeCommunity } = this.props;
    const { community } = match.params;
    if (community && activeCommunity !== community) {
      if (community === 'home') {
        this.props.history.push(`/${activeCommunity}/new`);
      } else actions.setCommunity(community);
    }

    if (location.pathname !== prevProps.location.pathname) {
      window.scrollTo(0, 0);
    }

    const userId = auth.user ? auth.user._id : null;
    const PrevUserId = prevProps.auth.user ? prevProps.auth.user._id : null;

    if (userId !== PrevUserId) {
      actions.userToSocket(userId);
    }

    if (!prevProps.auth.user && auth.user && !auth.user.webOnboard.onboarding) {
      actions.showModal('onboarding');
      actions.webOnboard('onboarding');
    }
  }

  toggleLogin() {
    this.setState({ openLoginModal: !this.state.openLoginModal });
  }

  closeModal() {
    this.props.history.push(this.props.location.pathname);
  }

  renderModal() {
    let { globalModal } = this.props;
    if (!globalModal) return null;
    globalModal = modals[globalModal] || globalModal;
    return (
      <Modal
        title={globalModal.title}
        header={globalModal.header}
        footer={globalModal.footer}
        close={() => this.props.actions.hideModal()}
        visible
      >
        {globalModal.body}
      </Modal>
    );
  }

  render() {
    const { location, user, match, children } = this.props;
    const temp = user && user.role === 'temp';
    const create = location.hash === '#newpost';
    const connectAccount = location.hash === '#connectAccount';

    let mobileEl = (
      <div className="mobileSplash">
        <h1>Relevant browser version doesn't currently support mobile devices</h1>
        <p>Please download a dedicated mobile app:</p>
        <p>
          <a
            href="https://itunes.apple.com/us/app/relevant-a-social-news-reader/id1173025051?mt=8"
            target="_blank"
          >
            <img alt="iOS App Store" src="https://relevant.community/img/appstore.png" />
          </a>
          &nbsp;&nbsp;&nbsp;&nbsp;
          <a
            href="https://play.google.com/store/apps/details?id=com.relevantnative&amp;pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1"
            target="_blank"
          >
            <img
              alt="Google Play Store"
              src="https://relevant.community/img/googleplaystore.png"
            />
          </a>
        </p>
      </div>
    );

    let header;
    if (location.pathname === '/') {
      header = <Header match={match} toggleLogin={this.toggleLogin.bind(this)} />;
      mobileEl = null;
    }

    return (
      <div>
        <GlobalStyle />
        <EthTools>
          {header}
          <div style={{ display: 'flex', width: '100%' }}>
            {/* <CommunityNav {...this.props} /> */}
            {children}
          </div>
          <AuthContainer
            toggleLogin={this.toggleLogin.bind(this)}
            open={this.state.openLoginModal || temp}
            modal
            {...this.props}
          />
          <Eth.Consumer>
            {wallet => (
              <AddEthAddress
                connectAccount={connectAccount}
                closeModal={this.closeModal.bind(this)}
                {...this.props}
                {...wallet}
              />
            )}
          </Eth.Consumer>
        </EthTools>
        <Modal
          title="New Post"
          visible={create}
          className="createPostModal"
          close={() => this.props.history.push(location.pathname)}
        >
          <CreatePost modal />
        </Modal>
        {this.renderModal()}
        <ToastContainer />
        {mobileEl}
        {renderRoutes(this.props.route.routes)}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  auth: state.auth,
  activeCommunity: state.community.active,
  navigation: state.navigation,
  globalModal: state.navigation.modal
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      ...navigationActions,
      ...authActions,
      getCommunities,
      getEarnings
    },
    dispatch
  )
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
