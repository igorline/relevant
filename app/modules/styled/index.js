import styled from 'styled-components/primitives';
import { mixins, sizing } from 'app/styles';

export const View = styled.View`
  ${mixins.margin}
  ${mixins.height}
  ${mixins.width}
  ${mixins.padding}
  ${mixins.background}
  ${mixins.borderRadius}
`;

export const Text = styled.Text`
  ${mixins.margin}
  ${mixins.height}
  ${mixins.width}
  ${mixins.padding}
  ${mixins.background}
  ${mixins.borderRadius}
`;

export const Image = styled.Image`
  height: ${sizing(3)};
  width: ${sizing(3)};
  ${mixins.margin}
  ${mixins.height}
  ${mixins.width}
  ${mixins.padding}
  ${mixins.background}
  ${mixins.borderRadius}
`;

export const ImageWrapper = styled.Text`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  ${mixins.margin}
`;
