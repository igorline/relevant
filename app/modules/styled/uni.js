import React, { useState, memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/primitives';
import { mixins, layout, fonts, colors, sizing, size, isNative } from 'app/styles';
import { abbreviateNumber } from 'utils/numbers';
import { TouchableOpacity } from 'react-native';
import ULink from 'modules/navigation/ULink.component';

export const Touchable = TouchableOpacity;

export const Box = styled.View`
  ${mixins.margin}
  ${mixins.padding}
`;

export const View = styled.View`
  ${mixins.margin}
  ${mixins.padding}
  ${mixins.flex}
  ${mixins.font}
  ${mixins.background}
  ${mixins.border}
  ${mixins.width}
  ${mixins.height}
  ${mixins.zIndex}
  ${mixins.borderRadius}
`;

export const AbsoluteView = styled.View`
  position: absolute;
  ${mixins.position}
`;

export const Text = styled.Text`
  ${mixins.margin}
  ${mixins.padding}
  ${mixins.flex}
  ${mixins.font}
  ${mixins.background}
  ${mixins.border}
  ${mixins.color}
  ${mixins.height}
`;

export const InlineText = styled.Text`
  ${mixins.margin}
  ${mixins.padding}
  ${mixins.font}
  ${mixins.color}
`;

export const Image = styled.Image`
  ${mixins.margin}
  ${mixins.height}
  ${mixins.width}
  ${mixins.padding}
  ${mixins.background}
  ${mixins.borderRadius}
  ${mixins.flex}
`;

export const ImageWrapper = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  ${mixins.margin}
  ${mixins.flex}
`;

export const FormImage = props => <Image {...props} {...layout.formImageProps} />;

export const Divider = styled.View`
  ${mixins.margin}
  ${mixins.padding}
  ${p => {
    if (p.screenSize && p.screenSize > 0) {
      return `
        height: ${sizing(4)};
        background-color: ${colors.dividerBg};
      `;
    }
    return layout.universalBorder('bottom');
  }}
`;

export const MobileDivider = styled(View)`
  height: ${sizing(4)};
  background-color: ${colors.dividerBg};
`;

export const Header = styled.Text`
  ${fonts.header}
  ${mixins.color}
  ${mixins.margin}
`;

export const Highlight = styled.Text`
  ${fonts.highlight}
  ${mixins.color}
  ${mixins.margin}
`;

export const Title = styled.Text`
  ${fonts.title}
  ${mixins.color}
  ${mixins.font}
  ${mixins.flex}
  ${mixins.margin}
`;

export const LinkFont = styled.Text`
  ${fonts.link}
  ${mixins.link}
  ${mixins.font}
  ${mixins.color}
 ${mixins.margin}
  ${() => (!isNative ? 'user-select: none; cursor: pointer;' : '')}
`;

export const CTALink = styled.Text`
  ${fonts.CTALink}
  ${mixins.link}
  ${mixins.color}
  ${mixins.font}
  ${mixins.margin}
`;

export const SecondaryText = styled.Text`
  ${fonts.secondaryText}
  ${mixins.color}
  ${mixins.font}
 ${mixins.margin}
`;

export const SmallText = styled.Text`
  ${fonts.secondaryText}
  color: ${colors.black};
  ${mixins.color}
  ${mixins.font}
 ${mixins.margin}
`;

export const AltLink = styled.Text`
  ${fonts.altLink}
  ${mixins.color}
  ${mixins.font}
  ${mixins.margin}
`;

export const CommunityLink = styled.Text`
  ${fonts.communityLink}
  ${mixins.color}
  ${mixins.font}
  ${mixins.margin}
`;

export const CommentText = styled.Text`
  ${fonts.commentText}
  ${mixins.color}
  ${mixins.font}
  z-index: 1;
`;

export const BodyText = styled.Text`
  ${fonts.bodyStyle}
  ${mixins.color}
  ${mixins.font}
  ${mixins.margin}
`;

export const StaticButton = styled(View)`
  ${layout.button}
  ${p =>
    p.disabled
      ? `
      background: ${colors.grey};
    & > div {
      color: ${colors.white};
    }
    `
      : ''};
  ${mixins.width}
  ${mixins.height}
  ${mixins.margin}
  ${mixins.background}
  ${mixins.padding}
  ${mixins.border}
  ${mixins.borderRadius}
  ${p => (p.hover && !p.active && !p.disabled ? layout.activeButtonShadow : '')}
  ${() => (!isNative ? 'user-select: none; cursor: pointer;' : '')}
`;

export const ButtonText = styled.Text`
  ${layout.buttonFont}
  ${mixins.color}
`;

HB.propTypes = {
  children: PropTypes.node,
  onPress: PropTypes.func,
  onClick: PropTypes.func
};

export const HoverButton = memo(HB);

function HB({ children, onPress, onClick, ...rest }) {
  const [hover, setHover] = useState(0);
  const [active, setActive] = useState(0);
  const isString = typeof children === 'string';
  const isArray = typeof children === 'object' && children.length;
  const isTextArray = isArray && children.find(el => typeof el === 'string');
  const renderString =
    !children || (!isArray && !children.$$typeof) || isTextArray || isString;
  return (
    <TouchableOpacity onClick={onClick} onPress={onPress}>
      <StaticButton
        hover={hover}
        active={active}
        onMouseDown={() => setActive(1)}
        onMouseUp={() => setActive(0)}
        onMouseEnter={() => setHover(1)}
        onMouseLeave={() => {
          setHover(0);
          setActive(0);
        }}
        {...rest}
      >
        {renderString ? <ButtonText {...rest}>{children}</ButtonText> : children}
      </StaticButton>
    </TouchableOpacity>
  );
}

export const Button = memo(HB);

ButtonWithIcon.propTypes = {
  text: PropTypes.string,
  image: PropTypes.node
};

export function ButtonWithIcon({ text, image, ...rest }) {
  return (
    <ViewButton fdirection={'row'} {...rest}>
      {image}
      <ButtonText>{text}</ButtonText>
    </ViewButton>
  );
}

export const ViewButton = styled(View)`
  ${layout.button}
  ${() => (!isNative ? 'cursor: pointer;' : '')}
  ${p =>
    p.disabled
      ? `
    background: ${colors.grey};
    `
      : ''};
  ${mixins.background}
  ${mixins.padding}
  ${mixins.margin}
`;

export const Tag = styled(Text)`
  ${layout.tag}
  ${p =>
    p.disabled
      ? `
    color: ${colors.grey};
    background: ${colors.secondaryBG};
    `
      : ''};
  ${mixins.background}
  ${mixins.padding}
  ${mixins.margin}
  ${mixins.color}
`;

export const NumericalValue = styled(Text)`
  ${fonts.numericalValue}
  ${mixins.width}
  ${mixins.font}
  ${mixins.inheritfont}
  ${mixins.color}
`;

export const BigNumber = styled(NumericalValue)`
  ${fonts.bigNumber}
`;

export const Spacer = styled(View)`
  flex-direction: row;
  position: relative;
  ${p => {
    if (Number.isInteger(p.nestingLevel) || Number.isInteger(p.additionalNesting)) {
      const total = (p.nestingLevel || 0) + (p.additionalNesting || 0);
      const UNIT = p.screenSize > 0 ? layout.NESTING_UNIT_RESONSIVE : layout.NESTING_UNIT;
      if (!total * UNIT) {
        return '';
      }
      return `padding-left: ${sizing((total - (p.screenSize > 0 ? 1 : 0)) * UNIT)};`;
    }
    return '';
  }}
  flex-grow: 1;
  ${mixins.flex}
`;

export const CloseX = styled(Image)`
  position: absolute;
  ${p => (p.position ? `position: ${p.position};` : null)}
  ${p => (p.top ? `top: ${size(p.top)};` : null)}
  ${p => (p.right ? `right: ${size(p.right)};` : null)}
  ${() => (!isNative ? 'user-select: none; cursor: pointer;' : '')}
  z-index: 10;
`;

export const Warning = styled(SmallText)`
  color: ${colors.warning};
`;

export const Overlay = styled(View)`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: ${colors.modalBackground};
`;

// export const Badge = styled(View)`
//   border-radius: 100%;
//   align-items: center;
//   justify-content: center;
//   display: flex;
//   flex-direction: row;
// `;
Badge.propTypes = {
  color: PropTypes.string,
  textColor: PropTypes.string,
  h: PropTypes.number,
  children: PropTypes.node,
  number: PropTypes.number
};

export function Badge({ color, textColor, h, children, number, ...styles }) {
  if (!number) return null;
  return (
    <View
      minwidth={h || 1.75}
      p={'0 0.5'}
      h={h || 1.75}
      bradius={(h || 1.75) / 2}
      align={'center'}
      justify={'center'}
      bg={color || colors.blue}
      {...styles}
    >
      <NumericalValue fs={1.25} lh={h || 1.75} c={textColor || colors.white}>
        {abbreviateNumber(number)}
      </NumericalValue>
    </View>
  );
}

export const Err = styled(Text)`
  color: ${colors.red};
`;

EthAddress.propTypes = {
  address: PropTypes.object
};

export function EthAddress({ address }) {
  if (!address) return null;
  return (
    <ULink to={`https://etherscan.io/address/${address}`} target="_blank" external>
      {address.slice(0, 6) + '...' + address.slice(address.length - 4, address.length)}
    </ULink>
  );
}

export const ErrorBox = ({ children, ...styleProps }) => (
  <View mt={2} p={2} bg={colors.errorA} border bc={colors.error} {...styleProps}>
    {children}
  </View>
);

export const WarningBox = ({ children, ...styleProps }) => (
  <View mt={2} p={2} bg={colors.warningA} border bc={colors.warning} {...styleProps}>
    {children}
  </View>
);

ErrorBox.propTypes = {
  children: PropTypes.node
};

WarningBox.propTypes = {
  children: PropTypes.node,
  styleProps: PropTypes.object
};
