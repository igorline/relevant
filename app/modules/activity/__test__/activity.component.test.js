import React from 'react';
import renderer from 'react-test-renderer';
import Activity from 'modules/activity/activity.component';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { auth, activity } from 'app/mockdata';
import { Text } from 'modules/styled/uni';

jest.mock('modules/post/web/post.component', () => 'PostComponent');
jest.mock('react-redux', () => ({ useDispatch: () => () => {}, connect: el => el }));

const props = {
  singleActivity: activity.activity,
  auth: auth.auth,
  actions: {
    goToProfile: jest.fn()
  },
  PostComponent: Text
};

test('Snapshot Activity Web', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Activity {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
