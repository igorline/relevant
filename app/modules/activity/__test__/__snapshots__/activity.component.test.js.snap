// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Activity Web 1`] = `
.c0 {
  color: #0000ff;
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "display": "flex",
      "maxWidth": "60rem",
      "position": "relative",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "justify",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "0rem",
        "marginLeft": "2rem",
        "marginRight": "2rem",
        "marginTop": "2rem",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "justify",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 1,
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexBasis": "0px",
          "WebkitFlexDirection": "row",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "0px",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0px",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignSelf": "flex-start",
            "alignSelf": "flex-start",
            "display": "flex",
            "marginRight": "0.75rem",
            "msFlexItemAlign": "start",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "display": "flex",
              "position": "relative",
            }
          }
        >
          <a
            className="c0 ulink-mock"
            href="/user/profile/bob"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitFlexBasis": "auto",
                  "borderBottomLeftRadius": "1rem",
                  "borderBottomRightRadius": "1rem",
                  "borderTopLeftRadius": "1rem",
                  "borderTopRightRadius": "1rem",
                  "display": "flex",
                  "flexBasis": "auto",
                  "height": "2rem",
                  "msFlexPreferredSize": "auto",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": "2rem",
                  "zIndex": 0,
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.00)",
                    "backgroundImage": "url(\\"https://yt3.ggpht.com/a/AGF-l7_IPKJORqk1Wo7AR4jMhW15Ma-za_QDlRxMcQ=s900-c-k-c0xffffffff-no-rj-mo\\")",
                    "backgroundPosition": "center",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "cover",
                    "bottom": "0px",
                    "height": "100%",
                    "left": "0px",
                    "position": "absolute",
                    "right": "0px",
                    "top": "0px",
                    "width": "100%",
                    "zIndex": -1,
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="https://yt3.ggpht.com/a/AGF-l7_IPKJORqk1Wo7AR4jMhW15Ma-za_QDlRxMcQ=s900-c-k-c0xffffffff-no-rj-mo"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "baseline",
            "WebkitBoxAlign": "baseline",
            "WebkitBoxDirection": "normal",
            "WebkitBoxFlex": 1,
            "WebkitBoxOrient": "vertical",
            "WebkitFlexBasis": "0px",
            "WebkitFlexDirection": "column",
            "WebkitFlexGrow": 1,
            "WebkitFlexShrink": 1,
            "alignItems": "baseline",
            "display": "flex",
            "flexBasis": "0px",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "msFlexAlign": "baseline",
            "msFlexDirection": "column",
            "msFlexNegative": 1,
            "msFlexPositive": 1,
            "msFlexPreferredSize": "0px",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxDirection": "normal",
              "WebkitBoxLines": "multiple",
              "WebkitBoxOrient": "horizontal",
              "WebkitBoxPack": "start",
              "WebkitFlexDirection": "row",
              "WebkitFlexWrap": "wrap",
              "WebkitJustifyContent": "flex-start",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "msFlexDirection": "row",
              "msFlexPack": "start",
              "msFlexWrap": "wrap",
              "position": "relative",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "marginRight": "0.375rem",
              }
            }
          >
            <a
              className="c0 ulink-mock"
              href="/user/profile/bob"
              hu={0}
              inline={1}
              onClick={[Function]}
              styles=""
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
                    "color": "rgba(0,0,255,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                    "textDecorationColor": "rgba(0,0,255,1.00)",
                  }
                }
              >
                @
                bob
              </div>
            </a>
          </div>
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.875rem",
                "lineHeight": "1.2249999999999999rem",
              }
            }
          >
            upvoted your comment
          </div>
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.875rem",
                "lineHeight": "1.2249999999999999rem",
              }
            }
          >
             → 
          </div>
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.875rem",
                "lineHeight": "1.2249999999999999rem",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406"
              dir="auto"
              style={
                Object {
                  "color": "rgba(0,0,0,1.00)",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": "0.875rem",
                  "lineHeight": "1.2249999999999999rem",
                }
              }
            >
              your reputation increased by 
              <span
                className="css-text-901oao css-textHasAncestor-16my406"
                dir="auto"
                style={
                  Object {
                    "WebkitTextDecorationColor": "rgba(126,211,33,1.00)",
                    "color": "rgba(126,211,33,1.00)",
                    "position": "relative",
                    "textDecorationColor": "rgba(126,211,33,1.00)",
                  }
                }
              >
                ▲
              </span>
               
              10%
            </span>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "display": "flex",
              "position": "relative",
            }
          }
        />
      </div>
    </div>
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(128,128,128,1.00)",
          "fontFamily": "HelveticaNeue",
          "fontSize": "0.75rem",
          "lineHeight": "1rem",
        }
      }
    >
      A Few Seconds Ago
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "display": "flex",
        "marginLeft": "3rem",
        "position": "relative",
      }
    }
  >
    <a
      className="c0 ulink-mock"
      href="/undefined/post/000000000000000000000003"
      hu={0}
      inline={0}
      onClick={[Function]}
      styles=""
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "display": "flex",
            "marginBottom": "0rem",
            "marginLeft": "2rem",
            "marginRight": "2rem",
            "marginTop": "0rem",
            "position": "relative",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "display": "flex",
              "position": "relative",
            }
          }
        />
      </div>
    </a>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(222,222,222,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
        "borderLeftStyle": "solid",
        "borderRightStyle": "solid",
        "borderTopStyle": "solid",
        "marginBottom": "0rem",
        "marginLeft": "0rem",
        "marginRight": "0rem",
        "marginTop": "1rem",
      }
    }
  />
</div>
`;
